colossus-metrics/src/main/scala/colossus/metrics/Aggregation.scala

```diff
@@ -38,20 +38,23 @@ object TagSelector {
 }
 
 sealed trait AggregationType {
-  def aggregate(values: Seq[Long]): Long
+  def aggregate(values: Seq[MetricValue]): RawMetricValue
 }
 object AggregationType {
   case object Sum extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum
+    def aggregate(values: Seq[MetricValue]): RawMetricValue = values.foldLeft(0L){case (build: RawMetricValue, next: MetricValue) => build + next.value}
   }
   case object Average extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum / values.size
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (build, next) => build + next.value} / values.size
   }
   case object Max extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.max
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (max, next) => if (max > next.value) max else next.value}
   }
   case object Min extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.min
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(Long.MaxValue){case (min, next) => if (min < next.value) min else next.value}
+  }
+  case object Natural extends AggregationType {
+    def aggregate(values: Seq[MetricValue]) = values.reduce{_ + _}.value
```

Comment:
must only be looking at the plus sign, compiler complains about no implicit `Numeric` when I try sum, just gonna leave it as is.